### PR TITLE
HA version 0.64.0 voluptuous config check.

### PR DESCRIPTION
**Pull request by JerryWorkman**
**Description:**

**Pull request in [home-assistant](https://github.com/home-assistant/home-assistant) (if applicable):** home-assistant/home-assistant#<home-assistant PR number goes here>

## Checklist:

- [ ] Branch: Fixes, changes and adjustments should be created against `current`. New documentation for platforms/components and features should go to `next`.
- [ ] The documentation follow the [standards][standards].

[standards]: https://home-assistant.io/developers/documentation/standards/

### CODE REVIEW

File: source/developers/development_validation.markdown [hunk 1]
```diff
@@ -28,6 +28,15 @@ Some things to keep in mind:
 - Use the constants defined in `const.py`
 - Import `PLATFORM_SCHEMA` from the parent component and extend it
 - Preferred order is `required` first and `optional` second
+- Starting with Home Assistant version 0.64.0 voluptuous now requires default values for config keys to be valid values. For instance:
+```
+vol.Required(CONF_SWITCH, default=None): cv.string,
```
Comment (fabaff) [inline source/developers/development_validation.markdown]: If a value is required then there shouldn't be a default. We want to fail if the value is missing.

**Review by fabaff — CHANGES_REQUESTED** [review]
According to the origin PR, it's Custom components only.

File: source/developers/development_validation.markdown (hunk 1 above)
Comment (JerryWorkman) [reply source/developers/development_validation.markdown]: Good point, but it was allowed with the previous version of Voluptuous. So the comment should be to remove the default parameter for vol.required but also insure that vol.optional has valid values.

**Comment by frenck** [comment]
Rebased this PR in order to fix the build.

**Review by fabaff — APPROVED** [review]

**Pull request merged**